lib/functions.php

```diff
@@ -762,6 +762,31 @@ function concat(array $iterators): Iterator
         return $emitter->iterate();
     }
 
+    /**
+     * Discards all remaining items and returns the number of discarded items.
+     *
+     * @template TValue
+     *
+     * @param Iterator $iterator
+     *
+     * @return Promise
+     *
+     * @psalm-param Iterator<TValue> $iterator
+     * @psalm-return Promise<int>
+     */
+    function discard(Iterator $iterator): Promise
+    {
+        return call(static function () use ($iterator) {
+            $count = 0;
+
+            while (yield $iterator->advance()) {
+                $count++;
```

Comment:
Based on what?

Reply:
Some benchmarks I saw a long time ago. Also it makes sense since postincrementation needs to preserve the original value for the expression where it's used.